applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <caf/events/ble_common_event.h>
+
+#include "usb_event.h"
+
+#define MODULE dvfs
+#include <caf/events/module_state_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DVFS_LOG_LEVEL);
+
+#include "ld_dvfs_handler.h"
+
+#define DVFS_RETRY_TIMEOUT		K_MSEC(500)
+/* Connection intervals used by LLPM are out of Bluetooth LE specification.
+ * The intervals are encoded by OR operation with a magic number of 0x0d00.
+ */
+#define REG_CONN_INTERVAL_LLPM_MASK	0x0d00
+
+enum dvfs_state {
+	DVFS_STATE_INITIALIZING,
+	DVFS_STATE_LLPM_CONNECTED,
+	DVFS_STATE_USB_CONNECTED,
+	DVFS_STATE_COUNT
+};
+
+/* Binary mask containing the frequency settings for each state.
+ * "1" means that the state requires the DVFS_FREQ_HIGH and "0" the DVFS_FREQ_MEDLOW.
+ * It is tailored for nrf54h20dk and might need to be adjusted for other platforms.
+ *
+ * DVFS_STATE_INITIALIZING	DVFS_FREQ_HIGH
+ * DVFS_STATE_LLPM_CONNECTED	DVFS_FREQ_MEDLOW
+ * DVFS_STATE_USB_CONNECTED	DVFS_FREQ_HIGH
+ */
+static uint8_t dvfs_freq_bitmask = 5;
```

Comment:
`BIT(DVFS_STATE_USB_CONNECTED) | BIT(DVFS_STATE_INITIALIZING)`

Reply:
Can we call it `dvfs_high_frequency_bitmask`? Also it should be `static const uint8_t` (it's configuration that is not expected to change)

Reply:
I wouldn't change name. Both names are not precise but current version is less misleading IMO

Reply:
DVFS integration uses 3 frequencies (high, medium, low). Because of that I think bitmask that elevates needed DVFS frequency to high needs to indicate high frequency in name. If not `dvfs_high_freq_bitmask`, then maybe we can call it `dvfs_high_freq_requests_bitmask` ?

Reply:
dvfs_high_medlow_freq_bitmask ?

Reply:
Discussed offline. We will use 2 separate bitmasks - one for high and the other one for medlow frequency